pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarStats.java

```diff
@@ -60,8 +60,8 @@
     private final ReentrantReadWriteLock bufferLock = new ReentrantReadWriteLock();
 
     public PulsarStats(PulsarService pulsar) {
-        this.topicStatsBuf = PooledByteBufAllocator.DEFAULT.heapBuffer(16 * 1024);
-        this.tempTopicStatsBuf = PooledByteBufAllocator.DEFAULT.heapBuffer(16 * 1024);
+        this.topicStatsBuf = Unpooled.buffer(16 * 1024);
```

Comment:
I know it won't save much in heap-memory but what if we use `directBuffer` here to save some heap-memory?

Reply:
I think we need the byte[] to write the result into an OutputStrean

Reply:
I see.